store/index.ts

```diff
@@ -39,3 +41,58 @@ export const currentWeekRangeStringAtom = atom((get) => {
 
   return `${startOfRange.toDateString()} - ${currentDate.toDateString()}`;
 });
+
+// NEW ATOMS
+export const dailyEventsWithTimesAtom = atom<EarthquakeEvent[] | undefined>(
+  undefined
+);
+
+export const allDailyEventsAtom = atom<Earthquakes | undefined>(undefined);
+
+export const processedDailyEventsWithTimesAtom = atom<
+  { hour: number; count: number }[] | undefined
+>((get) => {
+  const dailyEvents = get(dailyEventsWithTimesAtom);
+
+  if (!dailyEvents) return undefined;
+
+  return processEarthquakeData(dailyEvents);
+});
+
+const processEarthquakeData = (
+  events: EarthquakeEvent[]
+): { hour: number; count: number }[] => {
+  const hourlyCounts = Array(24).fill(0);
+
+  events.forEach((event) => {
+    const date = new Date(event.time);
+    const hour = date.getUTCHours(); // using UTC hour - we can handle converting later
+    hourlyCounts[hour]++;
+  });
+
+  return hourlyCounts.map((count, hour) => ({ hour, count }));
+};
+
+export const dailyActiveLocationsAtom = atom((get) => {
+  const dailyEvents = get(allDailyEventsAtom);
+  if (!dailyEvents) return undefined;
+  return getMostActiveLocations(dailyEvents);
+});
+
+// most active locations
+const getMostActiveLocations = (earthquakes: Earthquakes) => {
+  const regions: { [key: string]: number } = {};
+
+  earthquakes.forEach((event) => {
+    const region = event.properties?.place.split(', ').pop();
+
+    if (regions[region]) {
+      regions[region]++;
+    } else {
+      regions[region] = 1;
+    }
+  });
+
+  const sortedRegions = Object.entries(regions).sort((a, b) => b[1] - a[1]);
+  return sortedRegions.slice(0, 3).map((region) => region[0]);
+};
```

Comment:
There's a lot of atoms here (and routes in the fetchEarthquakes file) that are pretty much doing the same things - and making more API requests than we should really need to do it. 

This was mostly done because I wasn't sure *how* I was going to use the usgs API. But now that I have a better idea i need to establish:

- What values do I need for the WEEKLY and the DAILY dashboard items? 
- How can we distill this down into ONE API request for each 'set' of data points?
- Do we want these datapoints managed in a set of atoms (very atomic) or create a larger atom (fed from smaller atoms?) for each set of data points? e.g. a weeklyEventStats and dailyEventStats atoms 

A few other items I need to button up:
- The naming for everything listed above 
- For the "most active areas" we're using ALL events (no matter the magnitude), but for the other points we're using "significant magnitude" (2.5+ according to the USGS) - we probably should use the drilled down events for this section too? But does that make the main map confusing, when we see tons of little points in one area, but that area doesn't appear here?

Reply:
On this note too, I think it would make the sections easier to manage if we had DailyStatsSection and WeeklyStatsSection - that way we won't have to do all this `isWeekly ?` checking.

Reply:
All of this work will be captured in the next PR - see https://github.com/ash-bergs/earthquake-tracker/pull/10#issuecomment-2254642981